baseapp-chats/baseapp_chats/graphql/interfaces.py

```diff
@@ -0,0 +1,46 @@
+import graphene
+import swapper
+from baseapp_core.graphql import get_object_type_for_model
+from django.db.models import Sum
+from graphene import relay
+from graphene_django.filter import DjangoFilterConnectionField
+
+ChatRoom = swapper.load_model("baseapp_chats", "ChatRoom")
+UnreadMessageCount = swapper.load_model("baseapp_chats", "UnreadMessageCount")
+Block = swapper.load_model("baseapp_blocks", "Block")
+
+
+class ChatRoomsInterface(relay.Node):
+    chat_rooms = DjangoFilterConnectionField(get_object_type_for_model(ChatRoom))
+    unread_messages_count = graphene.Int()
+
+    def resolve_chat_rooms(self, info, **kwargs):
+        if not info.context.user.has_perm("baseapp_chats.list_chatrooms", self):
+            return ChatRoom.objects.none()
+
+        qs = ChatRoom.objects.filter(
+            participants__profile_id=self.pk,
+        ).order_by("-last_message_time", "-created")
+
+        # Exclude rooms where the profile is blocked
```

Comment:
Suggestion: `Exclude rooms blocked by the profile` and then `Exclude rooms where the profile is blocked`. Maybe change this single comment into two?